draft-ietf-mls-protocol.md

```diff
@@ -1971,15 +1985,24 @@ struct {
             Lifetime lifetime;
 
         case update:
-            struct{};
+            uint64 update_epoch;
 
         case commit:
+            uint64 commit_epoch;
             opaque parent_hash<V>;
     };
 
     Extension extensions<V>;
     /* SignWithLabel(., "LeafNodeTBS", LeafNodeTBS) */
     opaque signature<V>;
+    /* add_epoch is not authenticated */
+    select (LeafNode.leaf_node_source) {
+        case key_package:
+            uint64 add_epoch;
```

Comment:
I have another suggestion: the KeyPackage contains a LeafNode, which contains the `add_epoch` field. We can simply not care about it, or say that it MUST be equal to some fixed value such as `0`.
The `add_epoch` field is later modified by the person applying the Add, and that is not a problem because this field is not authenticated.
I think the integrity of this `add_epoch` field really has the same story as the integrity of the current unmerged leaves list: it is authenticated only by parent hash of nodes above it.